packages/web3-core/src/web3_context.ts

```diff
@@ -368,13 +368,14 @@ export class Web3Context<
 }
 
 // To avoid cycle dependency declare this type in this file
-export type TransactionBuilder<API extends Web3APISpec = unknown> = <
-	ReturnType = Transaction,
->(options: {
-	transaction: Transaction;
-	web3Context: Web3Context<API>;
-	privateKey?: HexString | Uint8Array;
-}) => Promise<ReturnType>;
+export type TransactionBuilder<API extends Web3APISpec = unknown> = <ReturnType = Transaction>(
+	options: {
+		transaction: Transaction;
+		web3Context: Web3Context<API>;
+		privateKey?: HexString | Uint8Array;
+	},
+	fillGasPrice: boolean,
```

Comment:
`fillGasPrice` should be apart of `options` argument, this keeps the function signature flexible to allow for the transaction builder to be overwritten without forcing any arguments that may not be needed by the implementation

Reply:
This should be applied every `fillGasPrice` was added in this PR